xml/operations-configuring-identity-websso.xml

```diff
@@ -840,4 +855,254 @@ keystone_sp_conf:
+    <itemizedlist>
+     <listitem>
+      <para>
+       If your cloud <emphasis role="bold">does not</emphasis> have AD FS
+       enabled, then replace the preceding
+       <literal>horizon_websso_choices:</literal> parameter with the following.
+      </para>
+<screen>- {protocol: openid, description: "OpenID Connect"}</screen>
+      <para>
+       The resulting block should look like the following.
+      </para>
+<screen>horizon_websso_choices:
+    - {protocol: openid, description: "OpenID Connect"}</screen>
+     </listitem>
+     <listitem>
+      <para>
+       If your cloud <emphasis role="bold">does</emphasis> have AD FS enabled,
+       then simply add the following parameter to the
+       <literal>horizon_websso_choices:</literal> section. Do not replace the
+       default parameter, add the following line to the existing block.
+      </para>
+<screen>- {protocol: saml2, description: "ADFS Credentials"}</screen>
+      <para>
+       If your cloud has AD FS enabled, the final block of your
+       <literal>~/openstack/ardana/ansible/roles/HZN-WEB/defaults/main.yml</literal>
+       should have the following entries.
+      </para>
+<screen>horizon_websso_choices:
+    - {protocol: openid, description: "OpenID Connect"}
+    - {protocol: saml2, description: "ADFS Credentials"}</screen>
+     </listitem>
+    </itemizedlist>
+   </step>
+   <step>
+    <para>
+     Run the following commands to add your changes to the local git
+     repository, and reconfigure the horizon service, enabling the changes made
+     in <emphasis role="bold">Step 1</emphasis>:
+    </para>
+<screen>cd ~/openstack
+git add -A
+git commit -m "Configured WebSSO using OpenID Connect"
+cd ~/openstack/ardana/ansible/
+ansible-playbook -i hosts/localhost config-processor-run.yml
+ansible-playbook -i hosts/localhost ready-deployment.yml
+cd ~/scratch/ansible/next/ardana/ansible
+ansible-playbook -i hosts/verb_hosts horizon-reconfigure.yml</screen>
+   </step>
+  </procedure>
+  </section>
+  <section>
+   <title>Create Identity Provider, Protocol, and Mapping</title>
+   <para>
+    To fully enable OpenID Connect, Identity Provider, Protocol, and Mapping for the given IdP must be created in Keystone. This is done by using the <emphasis role="bold">OpenStack Command Line Tool</emphasis> using the Keystone <emphasis role="bold">admin</emphasis> credential.
+   </para>
+   <orderedlist>
+    <listitem>
+     <para>
+      Log in to the &clm; node and source <literal>keystone.osrc</literal> file.
+     </para>
+<screen>source ~/keystone.osrc</screen>
+    </listitem>
+    <listitem>
+     <para>
+      Create the <emphasis role="bold">Identity Provider</emphasis>. For example:
+     </para>
+<screen>openstack identity provider create foo</screen>
+     <warning>
+      The name of the Identity Provider must be exactly the same as the "identity_provider" attribute given when configuring Keystone in the previous section.
+     </warning>
```

Comment:
This is causing a validation error. It should be 
```
<warning>
<para>
</para>
</warning>
```

Reply:
thanks for the info